src/main/scala/com/vngrs/etl/Pipeline.scala

```diff
@@ -0,0 +1,97 @@
+
+  def load(loaders: Loader[A]*): Unit = loaders.foreach(loader => loader(rdd))
+
+  def zip[B: ClassTag](other: Pipeline[B]): Pipeline[(A, B)] = new Pipeline(rdd.zip(other.rdd))
+
+  def union(other: Pipeline[A]): Pipeline[A] = new Pipeline(rdd.union(other.rdd))
+}
+
+object Pipeline {
+  def apply[A: ClassTag](extractor: Extractor[A])(implicit sc: SparkContext): Pipeline[A] = Pipeline(extractor(sc))
+
+  def apply[A: ClassTag](iterable: Iterable[A])(implicit sc: SparkContext): Pipeline[A] = Pipeline(sc.parallelize[A](iterable.toSeq))
+
+  def apply[A: ClassTag](arr: Array[A])(implicit sc: SparkContext): Pipeline[A] = Pipeline(sc.parallelize[A](arr))
+
+  def apply[A: ClassTag](rdd: RDD[A]): Pipeline[A] = new Pipeline(rdd)
+}
+
+private object UsageExamples {
+
+  implicit val sc: SparkContext = new SparkContext()
+
+  val extractor = new Extractor[String] {
+    override def apply(sc: SparkContext): RDD[String] = sc.parallelize[String](Seq("A", "B", "C"))
+  }
+
+  val transformer = new Transformer[String, String] {
+    override def apply(data: RDD[String]): RDD[String] = data.map(_.toLowerCase())
+  }
+
+  val squareTrans = new Transformer[Int, Double] {
+    override def apply(data: RDD[Int]): RDD[Double] = data.map(Math.pow(2, _))
+  }
+
+  val loader = new Loader[String] {
+    override def apply(data: RDD[String]): Unit = data.saveAsTextFile("path")
+  }
+
+  val extractorNum = new Extractor[Int] {
+    override def apply(sc: SparkContext): RDD[Int] = sc.parallelize[Int](Seq(1, 2, 3))
+  }
+
+  val transformerNum = new Transformer[(String, Int), String] {
+    override def apply(data: RDD[(String, Int)]): RDD[String] = data.map(t => s"${t._1}|${t._2}")
+  }
+
+  val loader2 = new Loader[String] {
+    override def apply(data: RDD[String]): Unit = data.saveAsTextFile("path")
+  }
+
+  final case class PowTrans(pow: Double) extends Transformer[Int, Double] {
+    override def apply(data: RDD[Int]): RDD[Double] = data.map(Math.pow(pow, _))
+  }
+
+  final case class FileLoader(path: String) extends Loader[String] {
+    override def apply(data: RDD[String]): Unit = data.saveAsTextFile(path)
+  }
+
+  Pipeline(extractor)
+    .transform(transformer)
+    .load(loader)
+
+  Pipeline(extractor)
+    .zip(Pipeline(extractorNum))
+    .transform(transformerNum)
+    .load(loader)
+
+  Pipeline(extractor)
+    .zip(Pipeline(extractorNum))
+    .transform(transformerNum)
+    .load(loader, loader2)
+
+  Pipeline(extractorNum)
+    .transform(PowTrans(2.0d))
+    .transform(Transformer.filter(_ > 15))
+    .zip(
+      Pipeline(extractorNum)
+        .transform(squareTrans)
+        .transform(Transformer.filter(_ > 3))
+    )
+    .transform(Transformer(_.toString))
+    .load(FileLoader("path/to/path"))
+
+  val pipeline1 = Pipeline(extractorNum)
+  val pipeline2 = Pipeline(extractorNum)
+
+  val zippedPipeline = Aggregator.zip(pipeline1, pipeline2)
```

Comment:
I believe having two different method is a nicer approach than having one method with a strategy enum.  So I prefer

```scala
Pipeline(...)
  .zip(Pipeline(..))
```

than

```scala
Pipeline(...)
  .combine(Pipeline(...), Strategy.Zip)
```

What I was trying to do is implementing `Combiner`s in different class as you suggested in your second code snippet. However this makes code ugly in my point of view. I still prefer first approach then;

```scala
trait Combiner
trait Zip extends Combiner {
  def apply(first: RDD, second: RDD) = first.zip(second)
}

Pipeline(...)
  .combine(Pipeline(...), Zip)
```

Yes, with this approach, `Combiner`s are extendible but syntax is as not nice as the first approach. Of course, this is just my opinion and if you guys saying that it is better to have something like this then I will implement in that way.

What in my mind is something like this;

```scala
trait Combiner
trait Zip extends Combiner {
  def apply ....
}

Pipeline(...)
  .combine(Zip(Pipeline(...)))
```

In order to do that, I believe I need partially applied function or something like that but I couldn't figure it out.

By the way, can you give me some example of usage of the akka stream's combine method?